src/topics/posts.js

```diff
@@ -287,4 +290,43 @@ module.exports = function (Topics) {
 
 		return returnData;
 	}
+
+	Topics.syncBacklinks = async (postData) => {
+		if (!postData) {
+			throw new Error('[[error:invalid-data]]');
+		}
+
+		// Scan post content for topic links
+		const matches = [...postData.content.matchAll(backlinkRegex)];
+		if (!matches) {
+			return 0;
+		}
+
+		const { pid, uid, tid } = postData;
+		let add = [];
+		await Promise.all(matches.map(async (match) => {
+			add.push(match[1]);
+		}));
```

Comment:
```
let add = [];
await Promise.all(matches.map(async (match) => {
	add.push(match[1]);
}));
```
Can replace that with
`let add = matches.map(match => match[1]);`

Reply:
Haha 👍 yes, most likely I was doing some sort of asynchronous call here before